src/WebJobs.Script/OutOfProc/Rpc/WebHostLanguageWorkerChannelManager.cs

```diff
@@ -102,16 +102,42 @@ public async Task SpecializeAsync()
         {
             _logger.LogInformation("Starting language worker channel specialization");
             _workerRuntime = _environment.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName);
+
             ILanguageWorkerChannel languageWorkerChannel = await GetChannelAsync(_workerRuntime);
             if (_workerRuntime != null && languageWorkerChannel != null)
             {
-                _logger.LogInformation("Loading environment variables for runtime: {runtime}", _workerRuntime);
-                await languageWorkerChannel.SendFunctionEnvironmentReloadRequest();
+                if (UsePlaceholderChannel(_workerRuntime))
+                {
+                    _logger.LogDebug("Loading environment variables for runtime: {runtime}", _workerRuntime);
+                    await languageWorkerChannel.SendFunctionEnvironmentReloadRequest();
+                }
+                else
+                {
+                    _logger.LogDebug("Shutting down placeholder worker. Worker is not compatible for runtime: {runtime}", _workerRuntime);
+                    // If we need to allow file edits, we should shutdown the webhost channel on specialization.
+                    await ShutdownChannelIfExistsAsync(_workerRuntime, languageWorkerChannel.Id);
+                }
             }
+
             _shutdownStandbyWorkerChannels();
             _logger.LogDebug("Completed language worker channel specialization");
         }
 
+        private bool UsePlaceholderChannel(string workerRuntime)
+        {
+            if (!string.IsNullOrEmpty(workerRuntime))
+            {
+                // Special case: node apps must be read-only to use the placeholder mode channel
+                // TODO: Remove special casing when resolving https://github.com/Azure/azure-functions-host/issues/4534
+                if (string.Equals(workerRuntime, LanguageWorkerConstants.NodeLanguageWorkerName))
+                {
```

Comment:
Same comment. Do we need to worry about casing?

Reply:
same here. this change is in dev already. we need to fix it